runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -1746,6 +1746,31 @@ static
 void genSuperClassTest(TR::Node *node, TR::Register *instanceClassReg, bool instanceClassRegCanBeReclaimed, TR::Register *castClassReg, int32_t castClassDepth,
                                             TR::LabelSymbol *falseLabel, TR_ARM64ScratchRegisterManager *srm, TR::CodeGenerator *cg)
    {
+
+   // When castClass is an unknown at compile time check if castClass is an interface or an array.
+   // If so skip superClassTest.
+   bool dynamicCastClass = castClassDepth == -1;
+   if ( dynamicCastClass )
+      {
+      TR::Register *scratchRegister1 = srm->findOrCreateScratchRegister();
+      TR::Register *scratchRegister2 = srm->findOrCreateScratchRegister();
+      TR_ASSERT(node->getSecondChild()->getOpCodeValue() != TR::loadaddr,
+            "genSuperClassTest: castClassDepth == -1 is not supported for a loadaddr castClass");
+
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmx, node, scratchRegister1, TR::MemoryReference::createWithDisplacement(cg, castClassReg, offsetof(J9Class, romClass)));
+      generateTrg1MemInstruction(cg, TR::InstOpCode::ldrimmw, node, scratchRegister1, TR::MemoryReference::createWithDisplacement(cg, scratchRegister1, offsetof(J9ROMClass, modifiers)));
+
+      // Array and interface check
+      static_assert(J9AccClassArray == 0x10000, "J9AccClassArray must be 0x10000");
+      static_assert(J9AccInterface == 0x200, "J9AccInterface must be 0x200");
```

Comment:
I don't think you need these two `static_assert()` lines when you load the value `(J9AccClassArray | J9AccInterface)` in a register using `loadConstant32()`.

Reply:
Removed the exact value asserts but added an assert to check that `(J9AccInterface | J9AccClassArray)` is a 32-bit value.